shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -88,64 +195,69 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             Append append = (Append) msg;
             Session session = setupSegments.get(new SimpleImmutableEntry<>(append.segment, append.getWriterId()));
```

Comment:
As per my current understanding entries are not removed from  `Map<Map.Entry<String, UUID>, Session> setupSegments `.  Since  `Session`  includes a `ByteBuf` this would have a non negligible size as there can be multiple sessions on a given connection.

Reply:
it actually takes 256 bytes minimal.

Reply:
The reason for my question here is based on my understanding that the underlying byte Arrays size once increased is never reduced. i.e.the underlying `io.netty.buffer.ByteBufAllocator#heapBuffer()` never reduces/frees the underlying byteArray until `io.netty.util.ReferenceCounted#release()` is invoked.  Also, `io.netty.buffer.ByteBuf#clear` simply sets the write index and read index to 0.

Reply:
yes, now, buffer release is invoked during flush